### DISCUSSION TITLE
Cryptic Mongoid Error Messages

### DISCUSSION THREAD

**Discussion by idbentley** [mongodb/mongoid#5395] [Q&A]
I've been working with Mongoid in Ruby on Rails for a week or two.  I have some experience with Rails, RoR and MongoDB independently, but this is the first time working with them together.

In the end this has been a frustrating experience because of the cryptic error messages that this library throws when using embedded models.

Specifically I'm referring to the error messages of the form:

`NameError: uninitialized constant EmbeddingModel::EmbeddedModel`

I'm hardly the first one to encounter this (for example: https://www.jamesridgway.co.uk/debugging-uninitialized-constant-issues-in-mongoid-active/).

There is no mention of this error in the docs, nor can I find any place where the fix of giving your `embeds_*` statement an explicit fully qualified class name anywhere in the docs.

Is there a good reason that Mongoid can't give a less cryptic error message?  Is there a reason this isn't documented?

**Comment by p-mongo** [comment]
Hi @idbentley ,

 `NameError` exception is raised by Ruby, not by Mongoid. As such the contents of this exception is not within Mongoid's control.

Mongoid does currently deviate from both Ruby and Rails behavior with respect to how it resolves association class names. This is described in detail [here](https://jira.mongodb.org/browse/MONGOID-5343). The current implementation in Mongoid offers better diagnostics in some cases but is problematic in others. We are planning to change Mongoid to match Ruby and Rails behavior.

**Reply by idbentley** [reply]
I understand what you are saying (though, I don't understand why the `NameError` is unavoidable, but that's ok).

I think it would be very helpful for new users of Mongoid if there was a "warning" in the documentation for the `embeds_*` functions.  If you agree, and can share for me the issue tracker for mongoid, I'll be happy to file a proper ticket.

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID
